### ISSUE TITLE
Favorites.m3u formatted incorrectly

### ISSUE THREAD

**Issue by duracell80** [moode-player/moode#73]
Spotted on this thread
http://moodeaudio.org/forum/showthread.php?tid=1925

> "What is a bit irritating for me though, is that I have added some additional radio stations via "New station" + button, which I later added to my list of Favorites. The issue is - that some of them has IP addresses as urls (or meaningless DNS names), so when having couple of such, it's becoming hard to navigate which one is which." Moode 6.3.0 and reproduced in 6.4.1.

For certain if a user is playing a user added station or even a moode default station and then adds the currently playing station to their favorites by way of clicking the heart icon in the player ... the resulting Favorites.m3u file is just a list of URL's without associated "track titles" or duration's (-1 for streams).

This then makes it harder to navigate the Favorites playlist as the list items can only be identified by their playback URL rather than a station name. The player doesn't seem to pick the user added station name but it will for a station that was in the build by default.

db entry looked fine in clean 6.4.1 adding HeartDance > https://media-ssl.musicradio.com/HeartDance, but station name was never added to Favorites.m3u
```
499|zx reserved 499|zx reserved 499|s|zx reserved 499
500|https://media-ssl.musicradio.com/HeartDance|Heart Dance|u|local
```

![faves](https://user-images.githubusercontent.com/18121278/73329137-0cd79e00-4222-11ea-9fb3-fe72c7292630.png)

![reproduced](https://user-images.githubusercontent.com/18121278/73329138-0cd79e00-4222-11ea-89af-556a303a1a2e.png)

M3U correct formatting ...
```
#EXTM3U

#EXTINF:-1,Gold 104.3
http://icy.arn.com.au/au_004_icy
```

**Comment by moodeaudio** [comment]
Regarding M3U formatting, the MPD command "save" is used. This command only writes the URL to the file because there is no metadata stored in MPD queue. It just contains URL's or file paths.

The issue where the station name from cfg_radio is not being displayed after add/edit a station or adding to favs is likely the result of RADIO.json array in playerlib.js not being updated. A page refresh will update the array but we want to make it so this is not needed. I'll have a look.

**Comment by monkeypox** [comment]
EXTM3U is actually an extension to M3U, see:

http://www.assistanttools.com/articles/m3u_playlist_format.shtml

We talked about this a little before. It’d be nice to support a more full featured format (.pls would also work and is what the radio stations currently use iirc) but as Tim said this is what mpd currently supports and doing something different requires someone to actually do it. :)

**Comment by moodeaudio** [comment]
After looking at the code it appears that indeed the session is not being updated when a station is deleted. This has caused an accumulation of deleted stations in the session file over time.

The fix will be included in 6.4.2 update.

**Comment by moodeaudio** [comment]
Closed. Fix was provided in 6.4.2
